### PR TITLE
only warn about io.EOF from mux.ReaderHeaders (fixes #1143)

### LINKED ISSUE #1143: Errors while running for a long time
Hi, this is going to be a very under-defined report, but when I run ipfs daemon v0.3.2 for a long time, I get this error sometimes:

    14:14:27.049 ERROR p2p/host/b: protocol mux failed: %sEOF basic_host.go:98

### PR DESCRIPTION
**Pull request by cryptix**
This should close #1143.

I've seen that error sometimes too and meant to fix it, if only for the '%s' not being used properly.

### CODE REVIEW

File: p2p/host/basic/basic_host.go [hunk 1]
```diff
@@ -95,7 +97,11 @@ func (h *BasicHost) newConnHandler(c inet.Conn) {
 func (h *BasicHost) newStreamHandler(s inet.Stream) {
 	protoID, handle, err := h.Mux().ReadHeader(s)
 	if err != nil {
-		log.Error("protocol mux failed: %s", err)
+		if err == io.EOF {
+			log.Warningf("protocol EOF: %s", s.Conn().RemotePeer())
```
Comment (jbenet) [inline p2p/host/basic/basic_host.go]: may even want it to just be Debug. a disconnection mid-stream is a common occurrence.

File: p2p/host/basic/basic_host.go [hunk 2]
```diff
@@ -95,7 +97,11 @@ func (h *BasicHost) newConnHandler(c inet.Conn) {
...
+		} else {
+			log.Errorf("protocol mux failed: %s", err)
```
Comment (jbenet) [inline p2p/host/basic/basic_host.go]: i think this should be dropped too, either warning, notice, or even just debug. end-users wont care about it, only devs/implementors.

(good to separate them though)

Comment (cryptix) [inline p2p/host/basic/basic_host.go]: I don't like to bury/ignore an error we have never seen.

Comment (jbenet) [inline p2p/host/basic/basic_host.go]: :+1:

Comment (whyrusleeping) [inline p2p/host/basic/basic_host.go]: > I don't like to bury/ignore an error we have never seen.

:100: :+1:

**Pull request merged**